src/Illuminate/Support/Str.php

```diff
@@ -379,6 +379,25 @@ public static function deduplicate(string $string, string $character = ' ')
         return preg_replace('/'.preg_quote($character, '/').'+/u', $character, $string);
     }
 
+    /**
+     * Truncates a string to a specified character limit and appends an ellipsis if necessary.
+     *
+     * @param  string  $value
+     * @param  int  $limit
+     * @param  string  $ellipsis
+     * @return string
+     */
+    public static function ellipsis(string $value, int $limit, string $ellipsis = '...'): string
+    {
+        if (mb_strlen($value) <= $limit) {
+            return $value;
+        }
+
+        $cut = $limit - mb_strlen($ellipsis);
+
+        return mb_substr($value, 0, $cut).$ellipsis;
```

Comment:
To be more precise, you'd need conditional logic here:

* When the last word is truncated, the ellipsis follow the word immediately without a space
* Is the last word the full, non-truncated word, however, the ellipsis are separated with a space

```suggestion
        $cut = $limit - (mb_strlen($ellipsis) + 1);
        $truncated = mb_substr($value, 0, $cut);
        $isFullWord = mb_substr($value, $cut, 1) === ' ';

        return $isFullWord ? $truncated . ' ' . $ellipsis : $truncated . $ellipsis;
```